src/api/layers/layers-external.api.js

```diff
@@ -0,0 +1,83 @@
+import axios from 'axios'
+import WmsCapabilities from '@/api/layers/wms-capabilities.class'
+import WmtsCapabilities from '@/api/layers/wmts-capabilities.class'
+import log from '@/utils/logging'
+
+/**
+ * Read and parse WMS GetCapabilities
+ *
+ * @param {string} baseUrl Base URL for the WMS server
+ * @returns {Promise<WmsCapabilities | null>} WMS Capabilities
+ */
+export async function readWmsCapabilities(baseUrl) {
+    const url = new URL(baseUrl)
+    url.searchParams.set('SERVICE', 'WMS')
+    url.searchParams.set('REQUEST', 'GetCapabilities')
+    const response = await axios.get(url.toString(), { timeout: 20000 })
```

Comment:
why the special timeout length?
might be good idea to set it as a constant on top of this file, if this is really needed, so that we declare this timeout only once

Reply:
I added a timeout to avoid having the spinner forever if the server is not answering, if got the axios docu correct by default there is no timeout !

Reply:
done